src/State.cpp

```diff
@@ -19,6 +19,19 @@
 #include "Streamline.h"
 #include "Upscaling.h"
 
+#define JSON_LOAD(setting_name, condition, json_data, action)                      \
```

Comment:
I see no reason this has to be a macro instead of a function. There are too much redundancy the way it is implemented now e.g. `json_data` is written again in the `condition`, and `setting_name` is basically included in `json_data` too.

IMO you should rewrite the signature as
```cpp
void json_load(json& base_json, std::string_view setting_name, std::function<bool(json&)> cond, std::function<void()> action)
```

Reply:
additionally, i think the condition is unncessary since you are already catching exceptions. The condition is inherent in value() function if u make it a template.

Reply:
I'll review but I was trying to keep the existing logic. I have no idea what it was doing other than crashing silently.

Reply:
Just realized I don't care enough about knowing what caused the error to rewrite this. Anyone can just see where the setting stopped and debug if it they want. It'll still rebuild due to catching the exception.